src/app.js

```diff
@@ -156,15 +157,74 @@ var run = function () {
   chromeCloudSync()
 
   // ----------------- editor ----------------------
-
   var editor = new Editor()
 
-  // ----------------- tabbed menu -------------------
+  // -------------- FileExplorer -------------------
+  var sources = {
+    'test/client/credit.sol': '',
+    'src/voting.sol': '',
+    'src/leasing.sol': '',
+    'src/gmbh/contract.sol': false,
+    'src/gmbh/test.sol': false,
+    'src/gmbh/company.sol': false,
+    'src/gmbh/node_modules/ballot.sol': false,
+    'src/ug/finance.sol': false,
+    'app/solidity/mode.sol': true,
+    'app/ethereum/constitution.sol': true
+  }
+  Object.keys(sources).forEach(function (key) { files.set(key, sources[key]) })
+
+  var fileExplorerAPI = {
+    filesList: function () { return Object.keys(files.list()) },
+    filesRemove: function (path) {
+      [...window.files.querySelectorAll('.file .name')].forEach(function (span) {
+        if (span.innerText === path) {
+          var li = span.parentElement
+          li.parentElement.removeChild(li) // delete tab
+        }
+      })
+      return files.remove(path)
+    },
+    filesIsReadOnly: function (path) { return files.isReadOnly(path) },
+    filesRename: function (oldPath, newPath) { return files.rename(oldPath, newPath) }
+  }
+  var api = new FileExplorer('#treeview', fileExplorerAPI, files.event)
+  api.on('focus', function (path) {
+    [...window.files.querySelectorAll('.file .name')].forEach(function (span) {
```

Comment:
oh, no. `window.files` is automatically set by the browser, and is a reference to the DOM element with the id `#files` - so i'm not defining it, i'm using it